Cargo.lock

```diff
@@ -110,6 +110,7 @@ dependencies = [
  "fish-printf",
  "libc",
  "lru",
+ "macro_rules_attribute",
```

Comment:
> I'm perfectly willing to squash+rebase if/when this is just about
 > ready to merge, but I'm keeping it as the original commits at
 > first on the off chance it helps whatsoever with reading the diff.

We don't treat pre-commit and post-commit review differently.
We optimize for both the same way. We're not gonna do a big-bang squash
but rather move (incrementally) towards a history that is good enough
and merge that.